tests/Microsoft.Identity.Test.Unit/PublicApiTests/CacheKeyExtensionTests.cs

```diff
@@ -0,0 +1,207 @@
+                                                              .WithHttpManager(httpManager)
+                                                              .WithExperimentalFeatures()
+                                                              .BuildConcrete();
+
+                await RunHappyPathTest(app, httpManager).ConfigureAwait(false);
+            }
+        }
+
+        [TestMethod]
+        public async Task CacheExtWithSerializedCacheTestAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                              .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                              .WithRedirectUri(TestConstants.RedirectUri)
+                                                              .WithClientSecret(TestConstants.ClientSecret)
+                                                              .WithHttpManager(httpManager)
+                                                              .WithExperimentalFeatures()
+                                                              .BuildConcrete();
+
+                app.AppTokenCache.SetBeforeAccess(BeforeCacheAccess);
+                app.AppTokenCache.SetAfterAccess(AfterCacheAccess);
+                await RunHappyPathTest(app, httpManager).ConfigureAwait(false);
+            }
+        }
+
+        private async Task RunHappyPathTest(ConfidentialClientApplication app, MockHttpManager httpManager)
+        {
+
+            var appCacheAccess = app.AppTokenCache.RecordAccess();
+
+            httpManager.AddInstanceDiscoveryMockHandler();
+            httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+            var result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                                    .WithAdditionalCacheKeyComponents(_additionalCacheKeys1)
+                                    .ExecuteAsync(CancellationToken.None)
+                                    .ConfigureAwait(false);
+
+            Assert.IsNotNull(result);
+            Assert.AreEqual("header.payload.signature", result.AccessToken);
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+            //Ensure that the order of the keys does not matter
+            result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                                .WithAdditionalCacheKeyComponents(_additionalCacheKeys3)
+                                .ExecuteAsync(CancellationToken.None)
+                                .ConfigureAwait(false);
+
+            Assert.IsNotNull(result);
+            Assert.AreEqual("header.payload.signature", result.AccessToken);
+            Assert.AreEqual(1, app.AppTokenCacheInternal.Accessor.GetAllAccessTokens().Count);
```

Comment:
You need to assert the exact cache key and the payload entries in the cache.

Reply:
Done